### ISSUE TITLE
Feature Request: Strict warning for implicit String/Int to Boolean assignments (even for valid literals)

### ISSUE THREAD

**Issue by sarath-mec** [okbob/plpgsql_check#196]
First off, I want to say this tool works great and I absolutely love it. It has been added to **Google CloudSQL** in the list of supported extensions, and it works like magic to capture errors in our environment.

I am filing this report because I noticed a specific behavior regarding implicit casting where the tool is silent on assignments that are technically valid at runtime but represent poor type strictness.

### Description
I have noticed that `plpgsql_check` does not issue warnings when assigning `INTEGER` or `TEXT` literals to a `BOOLEAN` variable, **provided that the values are valid PostgreSQL boolean constants**.

PostgreSQL natively supports implicitly casting the following to Boolean at runtime:
*   `VARCHAR` / `TEXT`: `'t'` or `'f'`
*   `VARCHAR` / `TEXT`: `'TRUE'` or `'FALSE'` (Case insensitive)
*   `VARCHAR` / `TEXT`: `'yes'` or `'no'` / `'y'` or `'n'`
*   `INTEGER`: `1` or `0`

**Observation:**
1.  If I assign an *invalid* string (e.g., `v_bool := 'random_text';`), `plpgsql_check` **correctly generates a warning** regarding invalid input syntax.
2.  If I assign a *valid* literal (e.g., `v_bool := 'true';` or `v_bool := 1;`), `plpgsql_check` remains **silent**.
3.  However, if I **RETURN** an Integer in a function defined to return Boolean, `plpgsql_check` **does warn**, regardless of the value.

I believe the tool is currently validating that the *value* is safe, but it is ignoring that the *types* are mismatched **during assignment**. I would expect `extra_warnings` or a strict mode to flag **all** implicit Text/Int-to-Boolean assignments to ensure code quality.

### Reproduction Code

```sql
CREATE OR REPLACE FUNCTION public.test_plpgsql_check_bugs()
 RETURNS boolean
 LANGUAGE plpgsql SECURITY DEFINER immutable
AS $function$
DECLARE
    v_bool_from_int     boolean;
    v_ret_int_from_bool integer;
BEGIN
    -- 1. Integer to Boolean Assignment
    -- ISSUE: No warning is generated here because 1 and 0 are valid boolean casts.
    v_bool_from_int := 1; -- Also works with 0

    -- 2. String to Boolean Assignment
    -- ISSUE: No warning is generated here because 'true' is a valid boolean literal.
    v_bool_from_int := 'true'; -- Works with 't', 'TRUE', 'y', 'yes', etc.
    
    -- Setup for return test
    v_ret_int_from_bool = 1;

    -- 3. Return Integer instead of Boolean
    -- CORRECT: A warning IS generated here ("target type is different type than source type").
    RETURN v_ret_int_from_bool; 
END;
$function$;
```

### Execution Check
```sql
SELECT * FROM plpgsql_check_function(
    'public.test_plpgsql_check_bugs()', 
    format:='json', 
    other_warnings:=TRUE, 
    extra_warnings:=TRUE, 
    fatal_errors:=FALSE, 
    performance_warnings:=TRUE, 
    security_warnings:=TRUE, 
    compatibility_warnings:=TRUE
);
```

### Actual Result
The tool only reports the issue on the `RETURN` statement. It ignores the assignments on the lines above it because the values (`1`, `'true'`) are valid constants.

### Expected Result
I expect `plpgsql_check` to output warnings for the assignments as well, indicating that a casting is happening implicitly. For example:
*   `assignment of "integer" value to "boolean" variable`
*   `assignment of "text" value to "boolean" variable`

***
